src/lib/pubkey/classic_mceliece/cmce_gf.h

```diff
@@ -0,0 +1,207 @@
+ * by the uint16_t 0b1101.
+ */
+class BOTAN_TEST_API Classic_McEliece_GF {
+   public:
+      /**
+       * @brief Creates an element of GF(q) from a uint16_t.
+       *
+       * Each element and the modulus is represented by a uint16_t, where the i-th least significant bit
+       * corresponds to the coefficient of z^i.
+       *
+       * @param elem The element as a uint16_t. Must be less than 2^m.
+       * @param modulus The modulus of GF(q).
+       */
+      Classic_McEliece_GF(CmceGfElem elem, CmceGfMod modulus) : m_elem(elem), m_modulus(modulus) {
+         BOTAN_DEBUG_ASSERT(elem <= (size_t(1) << log_q()) - 1);
+      }
+
+      /**
+       * @brief Get m.
+       *
+       * For a given irreducible polynomial @p modulus f(z) representing the modulus of a finite field GF(q) = GF(2^m),
+       * get the degree log_q of f(z) which corresponds to m.
+       *
+       * @param modulus The modulus of GF(q).
+       * @return size_t The degree log_q of the modulus (m for GF(2^m)).
+       */
+      static size_t log_q_from_mod(CmceGfMod modulus) { return floor_log2(modulus.get()); }
+
+      /**
+       * @brief Get m, the degree of the element's modulus.
+       *
+       * @return size_t The degree log_q of the modulus (m for GF(2^m)).
+       */
+      size_t log_q() const { return log_q_from_mod(m_modulus); }
+
+      /**
+       * @brief Get the GF(q) element as a GF_Elem.
+       *
+       * @return the element as a GF_Elem.
+       */
+      CmceGfElem elem() const { return m_elem; }
+
+      /**
+       * @brief Get the modulus f(z) of GF(q) as a GF_Mod.
+       *
+       * @return the modulus as a GF_Mod.
+       */
+      CmceGfMod modulus() const { return m_modulus; }
+
+      /**
+       * @brief Change the element to @p elem.
+       */
+      Classic_McEliece_GF& operator=(const CmceGfElem elem) {
+         m_elem = elem & CmceGfElem((size_t(1) << log_q()) - 1);
+         return *this;
+      }
+
+      /**
+       * @brief Divide the element by @p other in GF(q). Constant time.
+       */
+      Classic_McEliece_GF operator/(Classic_McEliece_GF other) const {
+         BOTAN_DEBUG_ASSERT(m_modulus == other.m_modulus);
+         return *this * other.inv();
+      }
+
+      /**
+       * @brief Add @p other to the element. Constant time.
+       */
+      Classic_McEliece_GF operator+(Classic_McEliece_GF other) const {
+         BOTAN_DEBUG_ASSERT(m_modulus == other.m_modulus);
+         return Classic_McEliece_GF(m_elem ^ other.m_elem, m_modulus);
+      }
+
+      /**
+       * @brief Add @p other to the element. Constant time.
+       */
+      Classic_McEliece_GF& operator+=(Classic_McEliece_GF other) {
+         BOTAN_DEBUG_ASSERT(m_modulus == other.m_modulus);
+         m_elem ^= other.m_elem;
+         return *this;
+      }
+
+      /**
+       * @brief XOR assign a GF_Elem to the element of this. Constant time.
+       */
+      Classic_McEliece_GF& operator^=(CmceGfElem other) {
```

Comment:
Shouldn't this be += keeping with the additive notation in GF(2)?

Reply:
Note that `CmceGfElem` is a strong type for uint16. If I remember correctly, we wanted to distinguish the operation with `Classic_McEliece_GF` objects from the operation with a `CmceGfElem`. However, while looking for its usage, I found that this operator is not used at all. Oops... I'll remove it ;)